### DISCUSSION TITLE
Retrieve signature from headers in rpc

### DISCUSSION THREAD

**Discussion by JulianSwales** [supabase/supabase#10127] [Questions] [answered]
I'm trying to setup an RPC function that is called by a mailersend webhook...  I'm trying to get the signature value from the header.  When I test the mailersend webhook against webhook.site the signature value is there in the header.  
So I have the following basic test rpc:
Create or replace function mailersend_webhook(data jsonb)
returns text
language plpgsql
security definer
as $$
declare
  t_signature text;
begin
  select current_setting('request.header.signature',true) into t_signature
  RAISE 'data: %', t_signature;
  return '200'
end;
$$

The data value does come through and contains the body of the request.
t_signature is always Null. Any suggestions? Any way to see all the header information? Better ways to do this? 
Thanks.

**Comment by GaryAustin1** [comment, marked as answer]
Try this format:
`SELECT current_setting('request.headers', true)::json->>'user-agent';`
from https://postgrest.org/en/stable/api.html#http-context
